### PR TITLE
PWN-5522 - Add log for app delete 

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5522

## Description of Work

[ what work was done in this ticket ]

## Screenshots (Optional)

Screenshot #1 | Screenshot #2 | Screenshot #3
:------------:|:-------------:|:------------:
[Paste Pic 1] | [Paste Pic 2] | [Paste Pic 3]

### CODE REVIEW

**Review by 1sakov — APPROVED** [review]

**Review by gslevinkov — APPROVED** [review]

**Comment by gslevinkov** [comment]
have you checked https://dev.appsflyer.com/hc/docs/install-android-sdk#backup-rules for backup rules?

**Comment by davran1001** [comment]
> have you checked https://dev.appsflyer.com/hc/docs/install-android-sdk#backup-rules for backup rules?

Yeah, but fullBackupContent is disabled in our app, there is no reason to provide prefs for storing this data

File: app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt [hunk 1]
```diff
@@ -6,26 +6,37 @@ import com.appsflyer.AppsFlyerConversionListener
 import com.appsflyer.AppsFlyerLib
 import com.google.firebase.messaging.RemoteMessage
 import org.p2p.wallet.BuildConfig
+import timber.log.Timber
+
+private const val UNINSTALL_APP_KEY = "af-uinstall-tracking"
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt:11]: KEY_UNINSTALL_APP

File: app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt [hunk 2]
```diff
@@ -6,26 +6,37 @@ import com.appsflyer.AppsFlyerConversionListener
...
+            Timber.tag(TAG).i("AppsFlyer service is initialized")
+        }.onFailure {
+            Timber.tag(TAG).i("AppsFlyer service, error on init  = $it")
         }
-        appsFlyer.init(devKey, listener, application)
-        appsFlyer.start(application)
     }
 
     fun onNewToken(newToken: String) {
+        val appsFlyerInstance = AppsFlyerLib.getInstance()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt:32]: to a field i meant to do as a property 

val appsFlyerInstance 
get() = AppsFlyerLib.getInstance()

File: app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt [hunk 3]
```diff
@@ -6,26 +6,37 @@ import com.appsflyer.AppsFlyerConversionListener
...
         // Sending new token to AppsFlyer
-        AppsFlyerLib.getInstance().updateServerUninstallToken(context, newToken)
+        appsFlyerInstance.updateServerUninstallToken(context, newToken)
         // the rest of the code that makes use of the token goes in this method as well
     }
 
     fun isUninstallTrackingMessage(message: RemoteMessage): Boolean {
-        return message.data.containsKey("af-uinstall-tracking")
+        val data = message.data
+        return data.containsKey(UNINSTALL_APP_KEY)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt:40]: previous was ok too

File: app/src/main/java/org/p2p/wallet/infrastructure/InfrastructureModule.kt [hunk 4]
```diff
@@ -100,5 +101,6 @@ object InfrastructureModule : InjectionModule {
         single { DefaultDispatchers() } bind CoroutineDispatchers::class
 
         single { PushTokenRepository() }
+        single { AppsFlyerService(androidContext()) }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/infrastructure/InfrastructureModule.kt:104]: singleOf

**Pull request merged**